buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -447,22 +449,17 @@ class BuildPlugin implements Plugin<Project> {
 
                 options.encoding = 'UTF-8'
                 options.incremental = true
-
-                if (project.javaVersion == JavaVersion.VERSION_1_9) {
-                    // hack until gradle supports java 9's new "--release" arg
-                    assert minimumJava == JavaVersion.VERSION_1_8
-                    options.compilerArgs << '--release' << '8'
-                }
+                options.compilerArgs << '--release' << project.targetCompatibility.majorVersion
```

Comment:
Can you add a comment to this line linking the relevant gradle issue for adding builtin support for this: https://github.com/gradle/gradle/issues/2510

Reply:
I pushed 9434717fb03d90667e0c7eba3eb2a250a9bc64eb.